src/daemon/daemon.cpp

```diff
@@ -2518,16 +2518,40 @@ try
         auto spec_it = vm_instance_specs.find(instance_name);
         assert(spec_it != vm_instance_specs.end() && "missing instance specs");
 
+        // TODO@snapshots need a way to query the existence of a snapshot before consuming it
+        try
+        {
+            auto snapshot = vm_ptr->get_snapshot(request->snapshot());
```

Comment:
You don't need this snapshot variable anywhere.

```suggestion
            vm_ptr->get_snapshot(request->snapshot());
```